prow/gitattributes/pattern.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitattributes
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	negativePrefix = "!"
+	patternDirSep  = "/"
+	zeroToManyDirs = "**"
+)
+
+// Pattern defines a single gitattributes pattern.
+type Pattern interface {
```

Comment:
Might make more sense to have this pattern in gitattributes.go where you define `Group` otherwise it isn't obvious who is consuming it